lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -200,6 +200,19 @@ sub update {
 
     my $entry = {};
     my ($error_message, $settings, $keys) = $self->_prepare_settings($table, $entry);
+
+    # Tables used in a group configured in YAML must not be modified manually
+    my $parameter;
+    $parameter = 'product_id' if $table eq 'Products';
+    $parameter = 'machine_id' if $table eq 'Machines';
+    $parameter = 'test_suite_id' if $table eq 'TestSuites';
```

Comment:
how about
```perl
my %param_from_table = (
    product_id => 'Products',
    machine_id => 'Machines',
    test_suite_id => 'TestSuites',
);
my $parameter = $param_from_table{$table};
```
Especially because you have this copy-pasted further down as well